src/utils/item.ts

```diff
@@ -130,3 +130,87 @@ export const paginationContentFilter = (
   items
     .filter((i) => i.type !== ItemType.FOLDER)
     .filter((i) => !i.settings?.isPinned);
+
+interface Node {
+  children: string[];
+}
+
+interface Tree {
+  [nodeId: string]: Node;
+}
+
+const createMapTree = (data: DiscriminatedItem[]) => {
+  const childrenTreeMap: Tree = {};
+  // eslint-disable-next-line no-plusplus
+  for (let i = 0; i < data.length; i++) {
+    const ele = data[i];
+    const parents = getParentsIdsFromPath(ele.path, { ignoreSelf: true });
+    const lastParent = parents[parents.length - 1];
+    if (lastParent) {
+      if (childrenTreeMap[lastParent]) {
+        childrenTreeMap[lastParent].children = [
+          ...childrenTreeMap[lastParent].children,
+          ele.id,
+        ];
+      } else {
+        childrenTreeMap[lastParent] = { children: [ele.id] };
+      }
+    }
+  }
+  return childrenTreeMap;
+};
+
+type ItemWithChildren = DiscriminatedItem & {
+  children: ItemWithChildren[];
+};
+
+interface TreeNode {
+  [nodeId: string]: ItemWithChildren;
+}
+
+// handle item children tree
+const buildItemsTree = (data: DiscriminatedItem[]) => {
+  const tree: TreeNode = {};
+  if (data.length === 1) {
+    tree[data[0].id] = { ...data[0], children: [] };
+  }
+  const mapTree: Tree = createMapTree(data);
+  const keys = Object.keys(mapTree);
+  const allChildren: string[] = keys.reduce((acc: string[], key: string) => {
+    const node = mapTree[key as keyof Tree];
+    if (node && node.children) {
+      acc.push(...node.children);
+    }
+    return acc;
+  }, []);
+  const rootKeys = keys.filter((key) => !allChildren.includes(key));
+
+  const buildTree = (nodeId: string) => {
+    const node = data.find((ele) => ele.id === nodeId);
+    if (mapTree[nodeId]) {
+      if (node) {
+        const nodeCopy = JSON.parse(JSON.stringify(node));
+        nodeCopy.children = mapTree[nodeId].children.map((childId) =>
+          buildTree(childId),
+        );
+        return nodeCopy;
+      }
+    }
+    return node;
+  };
```

Comment:
If you had the full children from the beginning you wouldn't need to find it again. Won't the following work?
I also supposed that you don't need a temp tree with `children` and that you use a new object to build your final tree.
```suggestion
  const buildTree = (node: DiscriminatedItem) => {
    if (mapTree[node.id]) {
        const entry = { 
          // you only need the name? or the id?
          id: node.id,
          name: node.name,
          children: mapTree[node.id].map((child) =>
            buildTree(child),
          );
        }
        return entry;
      
    }
    // we suppose you will always have a value so you should at least console here in case of errors
    console.error('node.id should have been in the map')
    return node
  };
```
Let me know if sth does not work with my suggestion.